examples/idx_bd_cache/main.go

```diff
@@ -104,27 +147,51 @@ func (plugin *ExamplePlugin) AfterInit() error {
 // Close is called by Agent Core when the Agent is shutting down. It is supposed
 // to clean up resources that were allocated by the plugin during its lifetime.
 func (plugin *ExamplePlugin) Close() error {
-	return safeclose.Close(plugin.Agent1, plugin.Agent2, plugin.Publisher, plugin.Agent1, plugin.Agent2,
-		plugin.bdIdxLocal, plugin.bdIdxAgent1, plugin.bdIdxAgent2, plugin.closeChannel)
+	return safeclose.Close(plugin.Agent1, plugin.Agent2, plugin.ETCDDataSync, plugin.Agent1, plugin.Agent2,
+		plugin.bdIdxLocal, plugin.bdIdxAgent1, plugin.bdIdxAgent2)
 }
 
 // Test data are published to different agents (including local).
 func (plugin *ExamplePlugin) publish() (err error) {
-	// Create bridge domain in local agent.
-	//br0 := l2tst.SimpleBridgeDomain1XIfaceBuilder("bd0", "iface0", true)
-	//err = plugin.Publisher.Put(l2.BridgeDomainKey(br0.Name), &br0)
-	//if err != nil {
-	//	return err
-	//}
-	//// Create bridge domain in agent1
-	//br1 := l2tst.SimpleBridgeDomain1XIfaceBuilder("bd1", "iface1", true)
-	//err = plugin.Agent1.Put(l2.BridgeDomainKey(br1.Name), &br1)
-	//if err != nil {
-	//	return err
-	//}
-	//// Create bridge domain in agent2
-	//br2 := l2tst.SimpleBridgeDomain1XIfaceBuilder("bd2", "iface2", true)
-	//err = plugin.Agent2.Put(l2.BridgeDomainKey(br2.Name), &br2)
+	//Create bridge domain in local agent.
+	br0 := &l2.BridgeDomains_BridgeDomain{
```

Comment:
We could make the example simpler by extracting creation of BD model using constructor with two parameters: bridge domain name, interface name(s).